x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/PolicyStepsRegistry.java

```diff
@@ -60,9 +63,11 @@ public class PolicyStepsRegistry {
     // keeps track of existing policies in the cluster state
     private final SortedMap<String, LifecyclePolicyMetadata> lifecyclePolicyMap;
     // keeps track of what the first step in a policy is, the key is policy name
-    private final Map<String, Step> firstStepMap;
+    private final Map<VersionedPolicyKey, Step> firstStepMap;
```

Comment:
We need the firstStep map versioned as if we the first action in a phase is reordered in a subsequent version we'd get a different first step in the phase depending on which version we're executing